TODO.md

```diff
@@ -0,0 +1,30 @@
+# To-do list (temporary)
+
+All items here are subject for discussion.
+
+## Tests
+
+* [x] Interference in nested scopes.
+* [x] Capturing for logger with level set by default.
+* [ ] No race (no records lost) when new records are appended in other thread while being pop-ed.
+* [x] `filter`/`drain` with each filters and combinations, including:
+    * [x] `name` is equal and parent, empty string is equivalent to `None`.
+    * [x] `message` from start, in the middle etc.
+    * [x] Different variants of `exc_info`.
+
+## Features
+
+### Filters
+
+* [x] By `level` (None, int, str), greater or equal (or `level` + `level_exact`? or `level_ge` + `level_eq`?).
+* [x] By `exc_info` (None, bool, class, tuple of classes or tuples).
+* [x] By `stack_info` (optional bool).
+
+### Other
+
+* [x] Better name for `find()`?
+* [ ] `assert_one_pop()` to pop matching and assert only one.  Is it possible to provide custome error message with similar records?   Or full list if it has limitted size.
+* [ ] Return pile from filters instead of list? Provide methods like `get_text()`?
+* [ ] Some way to automate `assert pile.empty()`? In `__exit__`? In `__del__`?
```

Comment:
`__exit__` may raise exceptions, `__del__` may not.

Reply:
And what is your opinion on the whole idea of such automation? I may conflict with other ideas like capturing for root logger by default (see below).

Reply:
Or may we'd better enable this behavior with some marker?

Reply:
It's not possible with fixture. Here is [a discussion](https://stackoverflow.com/a/48723067/168352) on philosophical aspect of this approach. But it should be possible with plugin hooks.